dask_cuda/benchmarks/local_cudf_merge.py

```diff
@@ -102,16 +103,25 @@ def get_random_ddf(chunk_size, num_chunks, frac_match, chunk_type):
         for i, part in enumerate(parts)
     }
 
-    return new_dd_object(graph, name, meta, divisions)
+    ddf = new_dd_object(graph, name, meta, divisions)
+
+    if chunk_type == "build":
+        if not args.no_shuffle:
+            divisions = [i for i in range(num_chunks)] + [num_chunks]
+            return ddf.set_index("shuffle", divisions=tuple(divisions))
+        else:
+            del ddf["shuffle"]
```

Comment:
Just so we're clear, calling `set_index` will be somewhat expensive.

If we're just generating random data then maybe we can generated it in an unordered way to start?  Or is there some particular dataset that we're using?  

If we do need to call `set_index` (which is about as expensive as a merge itself) then perhaps we should try to separate data generation from the benchmark with persist and wait calls?

Reply:
We might not need to call `set_index` here.  However, we **are** calling `persist` and `wait` before the merge time is measured

Reply:
OK, if that's the case then my concerns aren't that big of a deal.

Reply:
@mrcklin - You are correct that the `set_index`-based shuffle is nearly as expensive as the merge operation.   It may indeed make sense to either (1) separately report the set_index performance, or (2) be "smarter" about the data generation...  I will think a bit about this and make a note here.